processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
Allowing prototype-style inheritance on prototype methods could invite ambiguous matches in the AUTOMATIC mode. If we find some nice semantics for dealing with that (or living with it), we could surely relax that later on. Especially when we actually allow multi-level config-classes.